src/NServiceBus.Core/Recoverability/DefaultRecoverabilityPolicy.cs

```diff
@@ -18,9 +18,13 @@ public static class DefaultRecoverabilityPolicy
         /// <returns>The recoverability action.</returns>
         public static RecoverabilityAction Invoke(RecoverabilityConfig config, ErrorContext errorContext)
         {
-            if (errorContext.Exception is MessageDeserializationException)
+            // ReSharper disable once LoopCanBeConvertedToQuery
+            foreach (var unrecoverableExceptionType in config.Failed.UnrecoverableExceptionTypes)
```

Comment:
i would prefer the "inherits from" case rather than the "equals to"

i think this is more the more expected behavior.

perf  doesnt concern me. we are already in the scope of an exception being thrown. doing an inheritance check will be a factor smaller than the cost of .net building us a stack trace.